src/features/postsSlice.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-param-reassign */
+import { createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+
+type DefaultState = {
+  posts: Post[],
+  loaded: boolean,
+  hasError: boolean,
+};
+
+const initialState: DefaultState = {
+  posts: [],
+  loaded: false,
+  hasError: false,
+};
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    setLoading: (state, { payload }) => {
+      state.loaded = payload;
+    },
+    setHasError: (state, { payload }) => {
+      state.hasError = payload;
+    },
+    setPosts: (state, { payload }) => {
+      state.posts = payload;
+    },
+  },
+});
+
+export const { actions } = postsSlice;
```

Comment:
It`s the best practice, make it throughout your code
```suggestion
export const {  setLoading, setHasError, setPosts } = postsSlice.actions;
```